### DISCUSSION TITLE
Improve Revisions UI and Relationship / Modal UI

### DISCUSSION THREAD

**Discussion by TomSmedley** [payloadcms/payload#1440] [Roadmap]
At the moment, with Revisions, you get quite a technical layout to see the differences which is hard to understand for the client. It's using a raw value rather than a formatted value:
![image](https://user-images.githubusercontent.com/95056193/202463864-acc91ccc-ef2e-4923-9f71-9c09455e1642.png)

What has a really good revision experience is Contentful and maybe Payload can take from this experience.
![image](https://user-images.githubusercontent.com/95056193/202464310-875a4ea6-bf10-43b6-ad03-7c4f4bc74d10.png)

What it lets you do is choose which fields you want to roll back it also uses what is rendered on the edit screen. I'm thinking Payload can do something similar where rather than rendering the value, it renders a field as it would in the editor, however, it will be with `pointer-events: none` and be wrapped in a selectable box in which they can choose with value to carry over from the previous version. 

Here is a mockup of what I was thinking:
![mock-up](https://user-images.githubusercontent.com/95056193/202473786-3179b470-2ee8-4c63-9f75-8d248cb97e81.jpg)

----

I was also thinking using the remove icon from the upload field for the modal will allow for the same visual language.
![modal](https://user-images.githubusercontent.com/95056193/202516345-e41f2f50-ad78-4666-b366-ba01a09b3763.jpg)

----

This is just a personal one but I felt like the remove all button felt a little lost next to the drop down icon as one was an indication of the field being a drop down. By taking on the style of the "Add new" functionality, it becomes clear it's an actionable button:
![relationship](https://user-images.githubusercontent.com/95056193/202516669-716a5c3e-d987-40cb-ae59-691b996bcd25.jpg)

**Comment by JarrodMFlesch** [comment]
I think these are all great ideas.

**Comment by JarrodMFlesch** [comment]
Version diffs were updated in [this PR](https://github.com/payloadcms/payload/pull/12027)!
